docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+
+Some API changes would probably be required to make it suitable for use by UniFFI would also be necessary, but these would be tractable.
+
+We would need to update nimbus to use this client, and would need to move the client into the application-services repository, otherwise we would have issues with dependency management (ie, ensuring the upstream version changes continued to work for us, and managing the circular dependency which exists due to this library needing to use viaduct) and would complicate our build process (the library would need to be in our “megazord”, so pulling it remotely would complicate CI and our release process).
+These are the exact reasons why Nimbus itself is in the application-services repo.
+
+### Option 3: Use the existing nimbus client
+Splitting the [existing client](https://github.com/mozilla/application-services/blob/6069bd26d742f9597ab7e62d8df23ad634c7bd98/components/nimbus/src/client/http_client.rs#L9) out from Nimbus in a way that allows Nimbus to continue to use it, while also making it available for stand-alone use is also an attractive option.
+
+In particular, the feature set of that client overlaps with the requirements of the new library - no local persistence is necessary and no signature verification is required. It already works with Viaduct and is used by a component which is exposed via UniFFI.
+
+## Chosen Option
+
+We have chosen Option 3 because it allows us to reuse the new client in Nimbus, as well as on iOS and on Android with minimal initial development effort.
+If the new library ends up growing requirements that are already in the existing remote settings client, we remain able to copy that functionality from that library into this.
+
+## Specific Plans
+
+This section is non-normative - ie, is not strictly part of the ADR, but exists
+for context.
+
+This is a very high-level view of the tasks required here.
+
+* Create a new top-level component in the application-services repository (ie, under application-services/components), identify the exact API we wish to expose for this new library, describe this API using UniFFI, then implement the API with “stubs” (eg, having each method just be a rust todo!() macro or similar.)
+* Identify which parts of Nimbus should be factored out into a shared component (which would live under application-services/components/shared) and move that functionality to the new shared component. This component probably will not have a UniFFI .udl file, but is just for consumption by the new component above and the existing nimbus component (although this is up in the air slightly - if it is a requirement that nimbus and the new component share some configuration or initialization code, we might need to do something more complex here. This seems unlikely, but possible, so is included here for completeness). It is expected that a number of tests will also move here.
```

Comment:
nit: the brackets make this harder to read a little

Reply:
Can we simplify this with a diagram, so it's easier to follow what new components exists, which ones are exposed to the apps and which are library/shared ones

Reply:
I tried to simplify this and spent some time trying to find a reasonable way to represent this using Mermaid. I'm only partially happy with the result, so welcome feedback on whether it adds value or makes things worse! Pushed in 3be39fe19bee3b5faa714d7afe26e3dc097c0713

Reply:
I like it! It definitely adds value to me!